apps/system/utils/utils_ps.c

```diff
@@ -128,17 +133,14 @@ static void ps_print_values(char *buf, void *arg)
 		return;
 	}
 
-#ifdef CONFIG_SMP
-	cpu_idx = atoi(stat_info[PROC_STAT_CPU]);
-	if (cpu_idx != cpu && cpu != CONFIG_SMP_NCPUS) {
-		return;
-	}
-#endif
-	
-	printf("%5s | %4s | %4s | %7s | %c%c | %8s | %3s", stat_info[PROC_STAT_PID], stat_info[PROC_STAT_PRIORITY], \
+	printf("%5s | %4s | %4s | %7s | %c%c | %8s", stat_info[PROC_STAT_PID], stat_info[PROC_STAT_PRIORITY], \
 		flags & TCB_FLAG_ROUND_ROBIN ? "RR  " : "FIFO", utils_ttypenames[(flags & TCB_FLAG_TTYPE_MASK) >> TCB_FLAG_TTYPE_SHIFT], \
 		flags & TCB_FLAG_NONCANCELABLE ? 'N' : ' ', flags & TCB_FLAG_CANCEL_PENDING ? 'P' : ' ', \
-		utils_statenames[state], stat_info[PROC_STAT_CPU]);
+		utils_statenames[state]);
+
+#ifdef CONFIG_SMP
+	printf(" | %3s", stat_info[PROC_STAT_CPU]);
+#endif
```

Comment:
For SMP and for name, how about using the same way as cpuload?
I mean
1. printing all titles including cpu and name
2. when you print real value here, if cpu or name are disabled, then just print N/A

I wanna keep the code simple. It has too many conditionals now.

Reply:
done